bms-monitor.js

```diff
@@ -1,18 +1,28 @@
 const {expect} = require('chai');
+const Battery = require('./battery');
+const BMS_FIXED = require('./bmsConfig');
 
-function batteryIsOk(temperature, soc, charge_rate) {
-    if (temperature < 0 || temperature > 45) {
-        console.log('Temperature is out of range!');
-        return false;
-    } else if (soc < 20 || soc > 80) {
-        console.log('State of Charge is out of range!')
-        return false;
-    } else if (charge_rate > 0.8) {
-        console.log('Charge rate is out of range!');
-        return false;
+
+isPropertyOutOfRange = function(name,battery, bmsFixedValue) { 
+    let isOutOfRange = false;
+    if (battery[name] > bmsFixedValue.max || battery[name] < bmsFixedValue.min) {
+        console.log(`${bmsFixedValue.name} is out of range! - ${battery.printAllBatteryValues()}`);
+        isOutOfRange = true;
     }
-    return true;
+    return isOutOfRange;
+}
+
+function batteryIsOk(battery, bmsFixedValues) {
+   return !(isPropertyOutOfRange("temperature",battery,bmsFixedValues.temperature) || 

```

Comment:
When the number parameters grows, this statement gets very big.
Try javascript array `every` function to express the code. You may need to alter the function to `isPropertyInRange`, to make it easier.